app/src/keymap.c

```diff
@@ -179,20 +195,46 @@ int zmk_keymap_apply_position_state(int layer, uint32_t position, bool pressed,
         return err;
     }
 
-    if (pressed) {
-        return behavior_keymap_binding_pressed(&binding, event);
-    } else {
-        return behavior_keymap_binding_released(&binding, event);
+    enum behavior_locality locality = BEHAVIOR_LOCALITY_CENTRAL;
+    err = behavior_get_locality(behavior, &locality);
+    if (err) {
+        LOG_ERR("Failed to get behavior locality %d", err);
+        return err;
+    }
+
+    switch (locality) {
+    case BEHAVIOR_LOCALITY_CENTRAL:
+        return invoke_locally(&binding, event, pressed);
```

Comment:
Is there a reason this doesn't need to be wrapped in `#if ZMK_BLE_IS_CENTRAL`? Is it because if it's pressed on the peripheral, the position gets sent to the central and never gets sent back to be invoked?

Reply:
That's because technically, for a split or non-split, you'd always want to run this locally, and keymaps aren't ever processed anywhere else, so it doesn't need to be conditional in either case.

Reply:
Just clicked for me that the `#if ZMK_BLE_IS_CENTRAL` is being used to identify if the device is a split keyboard, not if it's the central half. Makes a lot more sense. Would `#if IS_ENABLED(CONFIG_ZMK_SPLIT)` be a better check since only central will ever be running it anyways?